plugin/plugin.go

```diff
@@ -90,22 +90,36 @@ package plugin // import "collectd.org/plugin"
 // derive_t  value_list_get_derive  (value_list_t *, size_t);
 // gauge_t   value_list_get_gauge   (value_list_t *, size_t);
 //
+// typedef int (*plugin_complex_config_cb)(oconfig_item_t);
+//
 // int wrap_read_callback(user_data_t *);
 //
 // int register_write_wrapper (char const *, plugin_write_cb, user_data_t *);
 // int wrap_write_callback(data_set_t *, value_list_t *, user_data_t *);
 //
 // int register_shutdown_wrapper (char *, plugin_shutdown_cb);
 // int wrap_shutdown_callback(void);
+//
+// int register_complex_config_wrapper (char *, plugin_complex_config_cb);
+// int validate_configs();
+// int process_complex_config(oconfig_item_t*);
+//
+// char *go_get_string_value(oconfig_item_t*, int);
```

Comment:
Noticed that configfile.h has similar definitions for getting value types from oconfig_item_t.